util/src/main/java/tc/oc/pgm/util/reflect/ReflectionUtils.java

```diff
@@ -25,4 +25,21 @@ public static Object readField(@Nullable Object obj, Field field) {
       field.setAccessible(wasAccessible);
     }
   }
+
+  public static void setField(Class<?> parent, Object base, Object value, String fieldName) {
+    try {
+      setField(base, value, parent.getDeclaredField(fieldName));
```

Comment:
Generally try to avoid repeated calls to get fields, from past experience java isn't very fast when creating `Field`s and it creates a new one each time you call. Prefer calling once to get the `Field` and reusing it, especially for things like the tablist where it already is a decent performance bottleneck from the amount of calls it can produce with high playercounts

Reply:
So would it be best to cache all of the fields used in the packet creation methods in NMSHacks?

Reply:
For all the fields that are called often, yeah